### PR TITLE
Adding App Search URI Protocol Handling

### PR DESCRIPTION
**Pull request by AgneLukoseviciute**
## Summary of the pull request
With this PR, if you win+R: ms-devhome:add-apps-to-cart?search="visual studio code"
You will be taken to this view of Dev Home:
image

## References and relevant issues
Related to (but does not close) https://github.com/microsoft/devhome/issues/2428

## Validation steps performed
Tested that the new URI protocol handling works and doesn't interfere with the other Activation handler for account/settings. 

## PR checklist
- [ ] Closes #xxx
- [ ] Tests added/passed
- [ ] Documentation updated

### CODE REVIEW

File: src/Services/ProtocolActivationHandler.cs [hunk 1]
```diff
@@ -19,6 +19,11 @@ public ProtocolActivationHandler(INavigationService navigationService)
         this._navigationService = navigationService;
     }
 
+    protected override bool CanHandleInternal(ProtocolActivatedEventArgs args)
+    {
+        return args.Uri != null && args.Uri.LocalPath.Equals(SettingsAccountsUri, StringComparison.OrdinalIgnoreCase);
```
[the remark this answers is not in the document]
Comment (AgneLukoseviciute) [reply src/Services/ProtocolActivationHandler.cs:24]: good catch, with the addition of CanHandleInternal we shouldn't need the check in HandleInternalAsync. I'll consolidate the two.

File: src/Services/ProtocolActivationHandler.cs [hunk 2]
```diff
@@ -19,6 +19,11 @@ public ProtocolActivationHandler(INavigationService navigationService)
...
     }
 
+    protected override bool CanHandleInternal(ProtocolActivatedEventArgs args)
+    {
+        return args.Uri != null && args.Uri.LocalPath.Equals(SettingsAccountsUri, StringComparison.OrdinalIgnoreCase);
+    }
+
     protected override Task HandleInternalAsync(ProtocolActivatedEventArgs args)
     {
         if (args.Uri.AbsolutePath == SettingsAccountsUri)
```
[the remark this answers is not in the document]
Comment (AgneLukoseviciute) [reply src/Services/ProtocolActivationHandler.cs:29]: will be resolved through consolidating this into a single check in CanHandleInternal

File: src/Services/AppInstallActivationHandler.cs [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    private static readonly char[] Separator = [','];
+
+    public AppInstallActivationHandler(
+        INavigationService navigationService,
+        SetupFlowViewModel setupFlowViewModel,
+        PackageProvider packageProvider,
+        IWindowsPackageManager wpm,
+        SetupFlowOrchestrator setupFlowOrchestrator)
+    {
+        this._navigationService = navigationService;
```
Comment (krschau) [inline src/Services/AppInstallActivationHandler.cs:40]: nit: remove `this.`

File: src/Services/ProtocolActivationHandler.cs [hunk 4]
```diff
@@ -19,14 +19,15 @@ public ProtocolActivationHandler(INavigationService navigationService)
...
-            _navigationService.DefaultPage = typeof(AccountsViewModel).FullName!;
-            _navigationService.NavigateTo(_navigationService.DefaultPage);
-        }
+        return args.Uri != null && args.Uri.AbsolutePath.Equals(SettingsAccountsUri, StringComparison.OrdinalIgnoreCase);
+    }
 
+    protected override Task HandleInternalAsync(ProtocolActivatedEventArgs args)
+    {
+        _navigationService.DefaultPage = typeof(AccountsViewModel).FullName!;
+        _navigationService.NavigateTo(_navigationService.DefaultPage);
```
Comment (krschau) [inline src/Services/ProtocolActivationHandler.cs:30]: I know this was existing code, but should we just make it
```suggestion
        _navigationService.NavigateTo(typeof(AccountsViewModel).FullName!);
```

File: src/Services/AppInstallActivationHandler.cs [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+    protected override bool CanHandleInternal(ProtocolActivatedEventArgs args)
+    {
+        return args.Uri != null && args.Uri.AbsolutePath.Equals(AppSearchUri, StringComparison.OrdinalIgnoreCase);
+    }
+
+    protected async override Task<Task> HandleInternalAsync(ProtocolActivatedEventArgs args)
+    {
+        await AppActivationFlowAsync(args.Uri.Query);
+        return Task.CompletedTask;
+    }
```
Comment (AmelBawa-msft) [inline src/Services/AppInstallActivationHandler.cs:56]:
```suggestion
    protected async override Task HandleInternalAsync(ProtocolActivatedEventArgs args)
    {
        await AppActivationFlowAsync(args.Uri.Query);
    }
```

**Review by AmelBawa-msft — APPROVED** [review]

**Review by dhoehna — APPROVED** [review]

**Comment by AgneLukoseviciute** [comment]
Thanks for the feedback, everyone! I don't have write permissions, but this should be good to merge now.

**Pull request merged**